### ISSUE TITLE
Error caused by missing framework.js when running gatsby develop

### ISSUE THREAD

**Issue by marozzocom** [Js-Brecht/gatsby-plugin-pnpm#8]
**Description**

When running `gatsby develop` framework code (e.g. React source) seems to be bundled in `commons.js` instead of `framework.js`, but Gatsby still attempts to load the file within a `<script>` element. The missing file leads to the 404 page thus causing an error in the console.

**To Reproduce**

1. Clone [example repository](https://github.com/marozzocom/gatsby-pnpm)
2. Run `pnpm install`
3. Run `npm run start`
4. Open developer tools and console

Expected:
No errors.

Observed:
Error in console: `Uncaught SyntaxError: Unexpected token '<'`

**Additional information**

Observed using
* Windows 10
* Node 16.4.2
* PNPM 6.11.5
* Gatsby 3.10.2
* gatsby-plugin-pnpm 1.2.6

This does not affect how the site works and does not seem to affect production builds at all, but it is annoying and led me to spend quite a bit of time investigating what was wrong in my configuration. It would be good if there was a way to fix the behavior, suppress the error or at least add a note to documentation.

I don't have enough knowledge to suggest this would be a "correct" fix, but changing the `framework` cacheGroup regexp seems to correct the issue, namely removing `(?<!node_modules.*)`.

```
// gatsby-node.js

const FRAMEWORK_BUNDLES = [`react`, `react-dom`, `scheduler`, `prop-types`]

exports.onCreateWebpackConfig = ({ actions }) => {
  actions.setWebpackConfig({
    optimization: {
      splitChunks: {
        cacheGroups: {
          framework: {
            test: new RegExp(
              `[\\\\/]node_modules[\\\\/](${FRAMEWORK_BUNDLES.join(
                `|`
              )})[\\\\/]`
            )}}}}})}
```

**Comment by Js-Brecht** [comment]
All this plugin does is tell Webpack where to look for modules and loaders.  The `cacheGroup` is configured in Gatsby core

**Comment by Js-Brecht** [comment]
So the problem is [here](https://github.com/gatsbyjs/gatsby/blob/15100915364ad6ea538d5384076260935e7967a6/packages/gatsby/src/utils/webpack.config.js#L545-L549)... the issue is that with pnpm, there's ALWAYS two levels of `node_modules`.

**Comment by Js-Brecht** [comment]
Sorry, I think I got a little bit ahead of myself.  It does make sense to add this configuration fix to this plugin instead of to the core.  I can add that soon

**Comment by Js-Brecht** [comment]
@marozzocom care to try `gatsby-plugin-pnpm@next`?

**Comment by Js-Brecht** [comment]
The fix I used here can actually be implemented in the core, which is a better idea.  That way there's only one source of truth for the framework bundles.  So I've opened https://github.com/gatsbyjs/gatsby/pull/32701

I can publish this for now though, assuming it works for you.  My tests were successful

**Comment by marozzocom** [comment]
After a very brief test, everything seems to be working, and the error is gone. 👍🏻 Thank you for the quick work, likewise for creating the PR in Gatsby -- I'll be following both to see where the change eventually goes!

**Comment by Js-Brecht** [comment]
Published in `gatsby-plugin-pnpm@1.2.7`
